src/host.rs

```diff
@@ -53,6 +53,15 @@ impl Peer {
         self.allowed_ips = allowed_ips;
     }
 
+    /// Resolves endpoint address to SocketAddr and sets the field
+    pub fn set_endpoint(&mut self, endpoint: &str) -> Result<(), WireguardInterfaceError> {
+        self.endpoint = match endpoint.parse() {
```

Comment:
```suggestion
        self.endpoint = Some(resolve(endpoint)?);
```

`.to_socket_addr()` i tak najpierw próbuje zrobić `.parse()`, więc nie potrzeba dublować
https://doc.rust-lang.org/src/std/net/socket_addr.rs.html#282-289